### PR TITLE
init: add import-service

### PR DESCRIPTION
**Pull request by Ivanko5417**
1 - File serverless.yml contains configuration for importProductsFile function
3 - The importProductsFile lambda function returns a correct response which can be used to upload a file into the S3 bucket
4 - Frontend application is integrated with importProductsFile lambda
5 - The importFileParser lambda function is implemented and serverless.yml contains configuration for the lambda

+1 - async/await is used in lambda functions
+1 - importProductsFile lambda is covered by unit tests (aws-sdk-mock can be used to mock S3 methods - https://www.npmjs.com/package/aws-sdk-mock)
+1 - At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder (move the file means that file should be copied into parsed folder, and then deleted from uploaded folder)

Link to import service: https://1ml9619jd3.execute-api.eu-west-1.amazonaws.com/dev
Link to FE PR: https://github.com/Ivanko5417/nodejs-aws-fe/pull/5
Link to frontend: https://d2iyewiu0ytduo.cloudfront.net/admin/products

### CODE REVIEW

File: import-service/src/handlers/importProductsFile.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { APIGatewayEventDefaultAuthorizerContext, APIGatewayProxyEventBase } from 'aws-lambda';
+import 'source-map-support/register';
+import { processError, processResponse } from '../utils/response.helper';
+import { ValidationError } from '../exceptions';
+import { getSignedUrl } from '../services/s3';
+
+export default async (event: APIGatewayProxyEventBase<APIGatewayEventDefaultAuthorizerContext>) => {
+  try {
+    if (!event.queryStringParameters || !event.queryStringParameters.name) {
```
Comment (mmarkelov) [inline import-service/src/handlers/importProductsFile.ts]: Can be simplified:
```js
if (!event.queryStringParameters?.name)
```

File: import-service/src/services/s3.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+export function getReadableStream(key) {
+  return s3.getObject({
+    ...defaultOptions,
+    Key: key,
+  }).createReadStream();
+}
+
+export async function moveObject(oldObjectKey: string, newPath: string) {
+  const objectName = oldObjectKey.split('/').pop();
+  const newObjectKey = newPath ? `${newPath}/${objectName}` : objectName;
```
Comment (mmarkelov) [inline import-service/src/services/s3.ts:29]: Do this check has any sense? As I can understand we should not do anything if `newPath` is not provided

Comment (Ivanko5417) [reply import-service/src/services/s3.ts:29]: that means that the file should be moved to root directory

**Comment by mmarkelov** [comment]
@Ivanko5417 could please provide some links to your Frontend PR? Also do you plan to add tests for **importProductsFile** function?

**Comment by Ivanko5417** [comment]
> @Ivanko5417 could please provide some links to your Frontend PR? Also do you plan to add tests for ** importProductsFile** function?

Sure, will be done during an hour

File: import-service/src/services/s3.ts (hunk 2 above)
Comment (mmarkelov) [reply import-service/src/services/s3.ts:29]: @Ivanko5417 understood

**Comment by Ivanko5417** [comment]
@mmarkelov I've added tests and updated PR description

**Comment by LoraBard** [comment]
@Ivanko5417  An error occurred when trying to upload .csv file [https://ibb.co/30BkY3P](https://ibb.co/30BkY3P)

**Comment by Ivanko5417** [comment]
> @Ivanko5417 An error occurred when trying to upload .csv file https://ibb.co/30BkY3P

content type of your file is wrong, it accepts only text/csv

**Pull request merged**